xla/stream_executor/stream_executor_internal.h

```diff
@@ -336,11 +336,48 @@ class StreamExecutorInterface {
   // Performs linear search over alive GPU streams.
   virtual Stream* FindAllocatedStream(void* /*gpu_stream*/) { return nullptr; }
 
+  // The following methods access an internal log of some subset
+  // of arguments passed to other class methods.
+  // Used for testing/debugging purposes.
+
+  struct GemmCallTrace {
+    enum class GemmType {
+      kPlain = 0,
+      kStridedBatched = 1,
+      kBatched = 2,
+      kBlasLt = 3
+    };
+    GemmType op;
+    int flags;
+    uint64_t size1, size2;
+  };
+  // This may be expanded as necessary to trace other calls
+  using ApiTrace = std::variant<GemmCallTrace>;
+
+  // Retrieves and clears internal argument logs.
+  virtual absl::StatusOr<std::vector<ApiTrace> > ExtractApiTrace() {
+    return absl::UnimplementedError("Not implemented");
+  }
+
+  // Bit mask
+  enum class SE_ArgumentLogging: uint64_t {
```

Comment:
Just name this ArgumentLoggingMode.  The "SE_" prefix is redundant, as you're already in the stream_executor namespace.